doc/man/8/ceph-objectstore-tool.rst

```diff
@@ -0,0 +1,472 @@
+
+Prerequisites
+^^^^^^^^^^^^^
+
+    * Having root access to the Ceph OSD node.
+    * Stopping the ceph-osd daemon. 
+
+Procedure
+^^^^^^^^^
+
+   Verify that the target OSD is down:
+
+   Syntax::
+
+    systemctl status ceph-osd@$OSD_NUMBER
+
+   Example::
+
+    [root@osd ~]# systemctl status ceph-osd@1
+
+   List the object’s attributes:
+
+   Syntax::
+
+    ceph-objectstore-tool --data-path $PATH_TO_OSD --pgid $PG_ID $OBJECT list-attrs
+
+   Example::
+
+    [root@osd ~]# ceph-objectstore-tool --data-path /var/lib/ceph/osd/ceph-0 --pgid 0.1c '{"oid":"zone_info.default","key":"","snapid":-2,"hash":235010478,"max":0,"pool":11,"namespace":""}' list-attrs
+
+
+MANIPULATING THE OBJECT ATTRIBUTE KEY
+-------------------------------------
+
+Use the ceph-objectstore-tool utility to change an object’s attributes. To manipulate the object’s attributes you need the data and journal paths, the placement group identifier (PG ID), the object, and the key in the object’s attribute.
+Note
+
+If using FileStore as the OSD backend object store, then add the `--journal-path $PATH_TO_JOURNAL` argument when getting, setting or removing the object’s attributes. Where the `$PATH_TO_JOURNAL` variable is the absolute path to the OSD journal, for example `/var/lib/ceph/osd/ceph-0/journal`.
+
+Prerequisites
+
+    * Having root access to the Ceph OSD node.  
+    * Stopping the ceph-osd daemon. 
+
+Procedure
+
+    Verify that the target OSD is down.
+
+ Syntax::
+
+    systemctl status ceph-osd@$OSD_NUMBER
+
+ Example::
+
+    [root@osd ~]# systemctl status ceph-osd@1
+
+ Get the object’s attributes:
+
+ Syntax::
+
+   ceph-objectstore-tool --data-path $PATH_TO_OSD --pgid $PG_ID $OBJECT get-attrs $KEY > $OBJECT_ATTRS_FILE_NAME
+
+ Example::
+
+   [root@osd ~]# ceph-objectstore-tool --data-path /var/lib/ceph/osd/ceph-0  --pgid 0.1c '{"oid":"zone_info.default","key":"","snapid":-2,"hash":235010478,"max":0,"pool":11,"namespace":""}' get-attrs "oid" > zone_info.default.attr.txt
+
+ Set an object’s attributes:
+
+ Syntax::
+
+   ceph-objectstore-tool --data-path $PATH_TO_OSD --pgid $PG_ID $OBJECT  set-attrs $KEY < $OBJECT_ATTRS_FILE_NAME
+
+ Example::
+
+   [root@osd ~]# ceph-objectstore-tool --data-path /var/lib/ceph/osd/ceph-0 --pgid 0.1c '{"oid":"zone_info.default","key":"","snapid":-2,"hash":235010478,"max":0,"pool":11,"namespace":""}' set-attrs "oid" < zone_info.default.attr.txt
+
+ Remove an object’s attributes:
+
+ Syntax::
+
+   ceph-objectstore-tool --data-path $PATH_TO_OSD --pgid $PG_ID $OBJECT rm-attrs $KEY
+
+ Example::
+
+   [root@osd ~]# ceph-objectstore-tool --data-path /var/lib/ceph/osd/ceph-0 --pgid 0.1c '{"oid":"zone_info.default","key":"","snapid":-2,"hash":235010478,"max":0,"pool":11,"namespace":""}' rm-attrs "oid"
+
+
+Options
+=======
+
+.. option:: --help          
+
+   produce help message
+
+.. option:: --type arg        
```

Comment:
This option is only needed if the tool can't tell the type from the --data-path.

Reply:
This information has been added to the man page.